### PR TITLE
Add rhel8 build dependencies

### PR DESCRIPTION
**Pull request by ntegan**
Added rhel8 build dependencies

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -18,13 +20,13 @@ RUN yum install -y \
     # Dependencies for building xen \
     checkpolicy gcc python38 python38-devel iasl ncurses-devel libuuid-devel glib2-devel \
     pixman-devel selinux-policy-devel yajl-devel systemd-devel \
-    glibc-devel.i686 glibc-devel \
+    glibc-devel.i686 glibc-devel flex bison wget \
```
Comment (ALSchwalm) [inline Dockerfile:23]: We need wget for building xen?

File: Dockerfile [hunk 2]
```diff
@@ -18,13 +20,13 @@ RUN yum install -y \
...
     \
     # Dependencies for building qemu \
     git libfdt-devel zlib-devel bzip2 \
     \
     # Crucible build dependencies \
-    rpm-build squashfs-tools openssl-devel rsync \
+    rpm-build squashfs-tools openssl-devel rsync python2 \
```
Comment (ALSchwalm) [inline Dockerfile:29]: I'm ok with this for now, but I'd like to move in the direction of killing any dependency we have on python2

File: Dockerfile [hunk 3]
```diff
@@ -6,6 +6,8 @@ LABEL maintainer="Adam Schwalm <adam.schwalm@starlab.io>"
 RUN yum update -y && yum install -y \
     # Add the dnf plugins so we can enable PowerTools \
     dnf-plugins-core \
+    # Needed for cpuid and systemd-networkd \
```
Comment (ALSchwalm) [inline Dockerfile]: Can you clarify that this is needed for the installroot, not for things we actually install in the container itself

File: Dockerfile (hunk 1 above)
Comment (ntegan) [reply Dockerfile:23]: It is in Xen's own build containers https://xenbits.xen.org/gitweb/?p=xen.git;a=blob;f=automation/build/archlinux/current.dockerfile;h=d46fc9d9ca76ba81e62732ee5eaab0b6bf6f9926;hb=HEAD

File: Dockerfile (hunk 2 above)
Comment (ntegan) [reply Dockerfile:29]: right now the main dependency that I've seen is download artifacts

File: Dockerfile (hunk 3 above)
Comment (ntegan) [reply Dockerfile]:
```
Curl error (37): Couldn't read a file:// file for file:///etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-8 [Couldn't open file /etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-8]
```

We need it in the container, as yum installroot epel-release followed by yum installroot systemd-networkd fails.
Yum looks for that rpm signing key in the build container even though we're using installroot

Comment (ALSchwalm) [reply Dockerfile]: Ah, sorry I was asking to updated the comment to explain that we don't need cpuid and systemd-networkd in the container, but rather for building stuff with it.

**Pull request merged**